oidc_setup.sh

```diff
@@ -0,0 +1,571 @@
+  log_error "Nextcloud admin username and password are required."
+  help
+  exit 1
+fi
+
+# Validate required configs for integration setup
+if [[ -z $NC_INTEGRATION_PROVIDER_TYPE ]] ||
+  [[ -z $SETUP_PROJECT_FOLDER ]] ; then
+  log_error "Following configs are required for integration setup:"
+  log_error "\tSETUP_PROJECT_FOLDER"
+  log_error "\tNC_INTEGRATION_PROVIDER_TYPE"
+  help
+  exit 1
+fi
+
+# Support for "Debug mode"
+if [[ $INTEGRATION_SETUP_DEBUG == "true"  ]]; then
+  log_info "Debug mode is enabled"
+  set -x
+  set -v
+fi
+
+if [[ $INTEGRATION_SETUP_TEMP_DIR == ""  ]]; then
+  log_info "Using default temp dir: ${INTEGRATION_SETUP_DEFAULT_TEMP_DIR}"
+  INTEGRATION_SETUP_TEMP_DIR=${INTEGRATION_SETUP_DEFAULT_TEMP_DIR}
+  mkdir -p ${INTEGRATION_SETUP_TEMP_DIR}
+else
+  log_info "Using ${INTEGRATION_SETUP_TEMP_DIR} as non-default temporary directory"
+  if [ ! -d "${INTEGRATION_SETUP_DEFAULT_TEMP_DIR}" ]; then
+    log_error "Temporary directory does not exist"
+    exit 1
+  fi
+fi
+
+# if something goes wrong or an error occurs during the setup of the whole integration
+# we can delete the storage created in OpenProject, otherwise it would need to be deleted manually when the script is run the next time
+deleteOPStorageAndPrintErrorResponse() {
+  log_error "Setup of OpenProject and Nextcloud integration failed when the OIDC Provider Type is '$NC_INTEGRATION_PROVIDER_TYPE'."
+  delete_storage_response=$(curl -s -o /dev/null -w "%{http_code}" -X DELETE -u${OP_ADMIN_USERNAME}:${OP_ADMIN_PASSWORD} \
+                      ${OPENPROJECT_BASEURL_FOR_STORAGE}/${storage_id} \
+                      -H 'accept: application/hal+json' \
+                      -H 'Content-Type: application/json' \
+                      -H 'X-Requested-With: XMLHttpRequest')
+  if [[ ${delete_storage_response} == 204 ]]; then
+      log_info "File storage name \"${OPENPROJECT_STORAGE_NAME}\" has been deleted from OpenProject!"
+  else
+      log_info "Failed to delete file storage \"${OPENPROJECT_STORAGE_NAME}\" from OpenProject!"
+  fi
+}
+
+# These URLs are just to check if the Nextcloud and OpenProject instances have been started or not before running the script
+NEXTCLOUD_HOST_STATE=$(curl -s -X GET ${NEXTCLOUD_HOST}/status.php)
+NEXTCLOUD_HOST_INSTALLED_STATE=$(echo $NEXTCLOUD_HOST_STATE | jq -r ".installed")
+NEXTCLOUD_HOST_MAINTENANCE_STATE=$(echo $NEXTCLOUD_HOST_STATE | jq -r ".maintenance")
+OPENPROJECT_BASEURL_FOR_STORAGE=${OPENPROJECT_HOST}/api/v3/storages
+openproject_host_state_response=$(curl -s -X GET -u${OP_ADMIN_USERNAME}:${OP_ADMIN_PASSWORD} ${OPENPROJECT_HOST}/api/v3/configuration)
+# These two data are set to "false" when the integration is done without project folder setup
+setup_project_folder=false
+setup_app_password=false
+# Minimum OpenProject version required in order to run this script.
+MINIMUM_OP_VERSION="14.2.0"
+# Openproject and Nextcloud Client name
+openproject_client_name="openproject"
```

Comment:
try to prefix the variable so that we can differentiate between NC and OP
```suggestion
NC_OIDC_OP_CLIENT_NAME="openproject"
```